packages/core/src/remoteconfig.ts

```diff
@@ -0,0 +1,146 @@
+import type {
+  RemoteOverrideableConfig,
+  RemoteConfigOptions,
+  RemoteConfigPayload,
+  RemoteConfigInterface,
+  RemoteConfigSource,
+} from '@sentry/types';
+
+const DEFAULT_CONFIG_NAME = '__default';
+
+/**
+ * Remote Configuration fetches configuration from a remote server.
+ */
+export function remoteConfig({
+  defaultConfigName = DEFAULT_CONFIG_NAME,
+  storage,
+  transport,
+}: RemoteConfigOptions): RemoteConfigInterface {
+  let _activeConfig: RemoteConfigPayload | undefined;
+  let _pendingConfig: RemoteConfigPayload | undefined;
+  let _lastFetch: Date | undefined;
+  let _lastUpdate: Date | undefined;
+  let _source: RemoteConfigSource = 'DEFAULT';
+  let _hasUpdate: boolean = false;
+  let _state: RemoteConfigState = 'INITIALIZING';
```

Comment:
I can't find the dfn of `RemoteConfigState` so far.

I think this state machine in here is going to be really important to get right because any of the state names we export will be hard to change later, and testing the different transitions will give the confidence that things are working well